src/main/java/net/imagej/ops/image/ascii/DefaultASCII.java

```diff
@@ -115,6 +117,8 @@ public String calculate(final IterableInterval<T> input) {
 			// normalized = (value - min) / (max - min)
 			tmp.set(cursor.get());
 			tmp.sub(min);
+			// NB: if a value is below min we set tmp to zero.
+			if (tmp.compareTo(zero) < 0) tmp.setZero();
```

Comment:
It is inconsistent to protect against <0 normalized values here, but then protect against >=1 normalized values inside the `charAt` call below. I would prefer to do both checks the same way: by clamping `charIndex` to `[0, charLen)`. Then you do not need any contortions with a special `zero` variable etc.